### PR TITLE
Use GeneratedDllImport in System.Diagnostics.EventLog

### PR DESCRIPTION
**Pull request by elinor-fung**
Also switched some of the p/invokes off of `StringBuilder`, since the generator doesn't support that.

@AaronRobinsonMSFT @jkoritzinsky

### CODE REVIEW

File: src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs [hunk 1]
```diff
@@ -321,10 +321,10 @@ public static object EvtGetEventInfo(EventLogHandle handle, UnsafeNativeMethods.
                 int error = Marshal.GetLastWin32Error();
                 if (!status)
                 {
-                    if (error == UnsafeNativeMethods.ERROR_SUCCESS)
+                    if (error == Interop.Errors.ERROR_SUCCESS)
                     { }
                     else
-                        if (error != UnsafeNativeMethods.ERROR_INSUFFICIENT_BUFFER)
+                        if (error != Interop.Errors.ERROR_INSUFFICIENT_BUFFER)
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs:327]: Can we fix the braces in this branch. It doesn't conform to the coding guidelines.

File: src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs [hunk 2]
```diff
@@ -663,58 +661,55 @@ public static void EvtSetChannelConfigProperty(EventLogHandle handle, UnsafeNati
...
-            sb.EnsureCapacity(ProviderIdNeeded);
-            status = UnsafeNativeMethods.EvtNextPublisherId(handle, ProviderIdNeeded, sb, out ProviderIdNeeded);
+            char[] buffer = new char[ProviderIdNeeded];
+            status = UnsafeNativeMethods.EvtNextPublisherId(handle, ProviderIdNeeded, buffer, out ProviderIdNeeded);
             win32Error = Marshal.GetLastWin32Error();
             if (!status)
                 EventLogException.Throw(win32Error);
 
-            return sb.ToString();
+            return new string(buffer, 0, ProviderIdNeeded - 1); // buffer includes null terminator
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs:712]: This pattern is used a lot. Is there any chance the returned size will be `0`?

File: src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/UnsafeNativeMethods.cs [hunk 3]
```diff
@@ -419,275 +363,247 @@ internal enum EvtSeekFlags
...
 
-        [DllImport(WEVTAPI)]
-        internal static extern bool EvtClose(IntPtr handle);
+        [GeneratedDllImport(Interop.Libraries.Wevtapi)]
+        internal static partial bool EvtClose(IntPtr handle);
 
         /*
-        [DllImport(WEVTAPI, EntryPoint = "EvtClose", SetLastError = true)]
+        [DllImport(Interop.Libraries.Wevtapi, EntryPoint = "EvtClose", SetLastError = true)]
         public static extern bool EvtClose(
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/UnsafeNativeMethods.cs:410]: Isn't this the same as above?

File: src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/UnsafeNativeMethods.cs [hunk 4]
```diff
@@ -419,275 +363,247 @@ internal enum EvtSeekFlags
...
+        [GeneratedDllImport(Interop.Libraries.Wevtapi, SetLastError = true)]
+        internal static partial EventLogHandle EvtOpenEventMetadataEnum(
                             EventLogHandle publisherMetadata,
-                            int flags
-                                    );
+                            int flags);
 
-        [DllImport(WEVTAPI, CharSet = CharSet.Auto, SetLastError = true)]
+        [GeneratedDllImport(Interop.Libraries.Wevtapi, SetLastError = true)]
         // public static extern IntPtr EvtNextEventMetadata(
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/UnsafeNativeMethods.cs:473]: Remove?

File: src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/UnsafeNativeMethods.cs (hunk 3 above)
Comment (elinor-fung) [reply src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/UnsafeNativeMethods.cs:410]: This is commented out (hit it with my find/replace). Will remove.

File: src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs (hunk 2 above)
Comment (elinor-fung) [reply src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs:712]: I'll add a check to return empty. I don't think it should be 0 (we always expect ERROR_INSUFFICIENT_BUFFER before).

**Review by jkoritzinsky — APPROVED** [review]

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**